### PR TITLE
[7.x] Updated ValidatesAttributes::validateUrl to use Symfony/Validator 5.1 regex

### PR DESCRIPTION
**Pull request by bsacharski**
The regex used by url validation logic was previously derived from version
2.7.4 of Symfony Validator component. As it turns out, old regex had a bug
where it wouldn't accept urls where domain was a hostname ending with a digit.
As such, the urls like http://domain1/ would be considered as invalid.

To fix this issue I've updated \Illuminate\Validation\Concerns\ValidatesAttributes::validateUrl
method using regex derived from version 5.1.0 of the Symfony Validator component.
To make sure that the fix works, new test cases were also added.

### CODE REVIEW

File: tests/Validation/ValidationValidatorTest.php [hunk 1]
```diff
@@ -2340,6 +2340,7 @@ public function testValidateEmailWithFilterCheck()
 
     /**
      * @dataProvider validUrls
+     * @group WIP
```
Comment (GrahamCampbell) [inline tests/Validation/ValidationValidatorTest.php:2343]: ?

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 2]
```diff
@@ -1706,24 +1706,26 @@ public function validateUrl($attribute, $value)
         }
 
         /*
-         * This pattern is derived from Symfony\Component\Validator\Constraints\UrlValidator (2.7.4).
+         * This pattern is derived from Symfony Validator component, file Constraints/UrlValidator.php (5.1.0).
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php:1709]: Please leave this as the class name.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php (hunk 2 above)
Comment (GrahamCampbell) [reply src/Illuminate/Validation/Concerns/ValidatesAttributes.php:1709]: There is no 5.1.0 version, also?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by bsacharski** [comment]
@GrahamCampbell thanks for review. I guess this is why I shouldn't code before going to bed and few minutes after waking up (rebased the master onto 7.x-derived branch...) :smile: 

Created new PR (with properly used & referenced STABLE Symfony version) #32307

**Pull request closed without merging**